datafusion/expr/src/expr.rs

```diff
@@ -297,7 +298,7 @@ pub enum Expr {
     /// [`ExprFunctionExt`]: crate::expr_fn::ExprFunctionExt
     AggregateFunction(AggregateFunction),
     /// Represents the call of a window function with arguments.
-    WindowFunction(WindowFunction),
+    WindowFunction(Box<WindowFunction>), // Boxed as it is large (272 bytes)
```

Comment:
This is the API change in this PR: it moves a "large" structure (272 bytes) on to the heap

the rest of the changes follow from this one change

Reply:
the concrete byte count is likely out-of-date soon, so I would just exclude it from the comment

Reply:
```suggestion
    WindowFunction(Box<WindowFunction>), // Boxed as it is large (> 272 bytes)
```